agent-control/src/cli/on_host/config_gen.rs

```diff
@@ -1,14 +1,311 @@
-use crate::cli::error::CliError;
+//! Implementation of the generate-config command for the on-host cli.
 
+use std::path::PathBuf;
+
+use tracing::info;
+
+use crate::{
+    cli::{
+        error::CliError,
+        on_host::config_gen::{
+            config::{AgentSet, AuthConfig, Config, FleetControl, Server, SignatureValidation},
+            identity::{Identity, provide_identity},
+            region::{Region, region_parser},
+        },
+    },
+    http::config::ProxyConfig,
+};
+
+pub mod config;
+pub mod identity;
+pub mod region;
+
+/// Generates the Agent Control configuration for host environments.
 #[derive(Debug, clap::Parser)]
-pub struct ConfigInputs {
+pub struct Args {
```

Comment:
I think there's some way to make arguments mutually exclusive to guarantee having either (parent ID + secret) or auth client ID but don't know for sure, did you check?

Reply:
Thanks for pointing it out! I though about it and then I totally forgot 😅 I'll dig into it!